orchestrator/test_runner/src/erc_721_happy_path.rs

```diff
@@ -0,0 +1,638 @@
+use crate::get_fee;
+use crate::utils::check_erc20_balance;
+use crate::utils::*;
+use crate::MINER_ADDRESS;
+use crate::MINER_PRIVATE_KEY;
+use crate::OPERATION_TIMEOUT;
+use crate::TOTAL_TIMEOUT;
+use bytes::BytesMut;
+use clarity::{Address as EthAddress, Uint256};
+use cosmos_gravity::query::get_attestations;
+use cosmos_gravity::send::send_to_eth;
+use cosmos_gravity::{query::get_oldest_unsigned_transaction_batches, send::send_ethereum_claims};
+use deep_space::address::Address as CosmosAddress;
+use deep_space::coin::Coin;
+use deep_space::private_key::PrivateKey as CosmosPrivateKey;
+use deep_space::Contact;
+use ethereum_gravity::utils::get_valset_nonce;
+use ethereum_gravity::{send_erc721_to_cosmos::send_erc721_to_cosmos, utils::get_tx_batch_nonce};
+use gravity_proto::gravity::query_client::QueryClient as GravityQueryClient;
+use gravity_proto::gravity::MsgSendToCosmosClaim;
+use gravity_proto::gravity::MsgValsetUpdatedClaim;
+use gravity_utils::error::GravityError;
+use gravity_utils::types::SendToCosmosEvent;
+use num::CheckedAdd;
+use prost::Message;
+use std::any::type_name;
+use std::time::Duration;
+use std::time::Instant;
+use tokio::time::sleep as delay_for;
+use tonic::transport::Channel;
+use web30::client::Web3;
+
+pub async fn erc721_happy_path_test(
+    web30: &Web3,
+    grpc_client: GravityQueryClient<Channel>,
+    contact: &Contact,
+    keys: Vec<ValidatorKeys>,
+    gravity_address: EthAddress,
```

Comment:
you need the GravityERC721 address here, as well as the ERC721 contract deployed for testing.

You should remove the erc20 address as it shouldn't be needed.

Reply:
Updated: 
```
pub async fn erc721_happy_path_test(
    web30: &Web3,
    contact: &Contact,
    keys: Vec<ValidatorKeys>,
    gravity_address: EthAddress,
    gravity_erc721_address: EthAddress,
    erc721_address: EthAddress,
    validator_out: bool,
) 
```